lib/rom/sql/extensions/rails_log_subscriber.rb

```diff
@@ -15,20 +15,31 @@ def sql(event)
         binds = payload[:binds].to_a.inspect if payload[:binds]
 
         if odd?
-          name = color(name, :cyan, true)
-          sql  = color(sql, nil, true)
+          name = color(name, :cyan, color_option)
+          sql  = color(sql, nil, color_option)
         else
-          name = color(name, :magenta, true)
+          name = color(name, :magenta, color_option)
         end
 
         debug "  #{name}  #{sql}  #{binds}"
       end
 
       attr_reader :odd_or_even
       private :odd_or_even
+
       def odd?
         @odd_or_even = !odd_or_even
       end
+
+      private
+
+      def color_option
+        if Gem::Version.new(Rails::VERSION::STRING) >= Gem::Version.new('7.1')
+          { color: true }
+        else
+          true
+        end
```

Comment:
I have two suggestions

1. You can assign it to a constant `COLOR_OPTION = if ...` will work just as fine but with only 1 check rather than a check per every call
2. You use `Rails` constant but the plugin depends on active-support, you should check its version instead. For instance, I have a project where active-support is present but rails isn't.